src/Strategy/Maven/Pom.hs

```diff
@@ -64,19 +65,30 @@ data MavenLabel
   | MavenLabelOptional Text
   deriving (Eq, Ord, Show)
 
-toDependency :: MavenPackage -> Set MavenLabel -> Dependency
-toDependency (MavenPackage group artifact version) = foldr applyLabel start
+toDependency :: MavenPackage -> Set MavenLabel -> MavenDependency
+toDependency (MavenPackage group artifact version) = foldr applyLabelToMavenDep start
   where
-    start :: Dependency
-    start =
-      Dependency
-        { dependencyType = MavenType
-        , dependencyName = group <> ":" <> artifact
-        , dependencyVersion = CEq <$> version
-        , dependencyLocations = []
-        , dependencyEnvironments = mempty
-        , dependencyTags = Map.empty
-        }
+    start :: MavenDependency
+    start = do
+      let dep =
+            Dependency
+              { dependencyType = MavenType
+              , dependencyName = group <> ":" <> artifact
+              , dependencyVersion = CEq <$> version
+              , dependencyLocations = []
+              , dependencyEnvironments = mempty
+              , dependencyTags = Map.empty
+              }
+      MavenDependency dep (Set.fromList [])
+
+    applyLabelToMavenDep :: MavenLabel -> MavenDependency -> MavenDependency
+    applyLabelToMavenDep lbl mavenDep = do
+      case lbl of
+        MavenLabelScope scope ->
+          mavenDep{dependency = depWithLabel, dependencyScopes = Set.insert scope (dependencyScopes mavenDep)}
+        _ -> mavenDep{dependency = depWithLabel}
+      where
+        depWithLabel = applyLabel lbl (dependency mavenDep)
```

Comment:
[Prefer, Optional] provide type inference for readability

Reply:
Removing function in where clause so no need to provide type inference here